src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2804,14 +2804,6 @@ private static void CheckTupleElementNames(int cardinality, ImmutableArray<strin
                 {
                     throw new ArgumentException(CodeAnalysisResources.TupleNamesAllOrNone, nameof(elementNames));
                 }
-
```

Comment:
On second thought, I should still have a check here that some of the element names are non-null.